src/pykeen/nn/node_piece.py

```diff
@@ -222,7 +228,11 @@ def __call__(self, edge_index: numpy.ndarray) -> numpy.ndarray:  # noqa: D102
             x_old = x
         if no_convergence:
             logger.warning(f"No covergence after {self.max_iter} iterations with epsilon={self.epsilon}.")
-        return numpy.argpartition(x, max(x.size - self.num_anchors, 0))[-self.num_anchors :]
+
+        nodes = numpy.argpartition(x, max(x.size - self.num_anchors, 0))
+        unique_anchors = nodes[~numpy.isin(nodes, known_anchors)][-self.num_anchors:]
+        unique_anchors = numpy.concatenate([known_anchors, unique_anchors])
+        return unique_anchors
```

Comment:
cf. above. Maybe put this into a helper method, too, since the code looks very similar.